core-primitives/sgx/crypto/src/rsa3072.rs

```diff
@@ -0,0 +1,94 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::sgx_reexport_prelude::*;
+
+use crate::{
+	error::{Error, Result},
+	traits::ShieldingCrypto,
+};
+use derive_more::Display;
+use itp_settings::files::RSA3072_SEALED_KEY_FILE;
+use itp_sgx_io::{seal, unseal, SealedIO};
+use log::*;
+use sgx_crypto_helper::{
+	rsa3072::{Rsa3072KeyPair, Rsa3072PubKey},
+	RsaKeyPair,
+};
+use std::{sgxfs::SgxFile, vec::Vec};
+
+#[derive(Copy, Clone, Debug, Display)]
+pub struct Rsa3072Seal;
+
+impl SealedIO for Rsa3072Seal {
+	type Error = Error;
+	type Unsealed = Rsa3072KeyPair;
+	fn unseal() -> Result<Self::Unsealed> {
+		let raw = unseal(RSA3072_SEALED_KEY_FILE)?;
+		let key: Rsa3072KeyPair =
+			serde_json::from_slice(&raw).map_err(|e| Error::Other(format!("{:?}", e).into()))?;
+		Ok(key.into())
+	}
+
+	fn seal(unsealed: Rsa3072KeyPair) -> Result<()> {
+		let key_json = serde_json::to_vec_pretty(&unsealed)
```

Comment:
I see, as the integration tests run, I think this works. I still feel a bit uneasy with the `_pretty` suffix. I think it is better to remove it. We don't gain anything with this.